pkgs/applications/networking/cluster/nomad/default.nix

```diff
@@ -23,11 +24,18 @@ let
         inherit sha256;
       };
 
+      nativeBuildInputs = [ installShellFiles ];
+
       # ui:
       #  Nomad release commits include the compiled version of the UI, but the file
       #  is only included if we build with the ui tag.
       tags = [ "ui" ];
 
+      postInstall = ''
+        echo "complete -C $out/bin/nomad nomad" > nomad.bash
+        installShellCompletion nomad.bash
+      '';
```

Comment:
I just copied this from how Vault installs bash completion:

https://github.com/NixOS/nixpkgs/blob/1eaf23e8febf68394df1e6d3304c7a02dbf4c59c/pkgs/tools/security/vault/default.nix#L33-L35